android/src/main/java/android/print/PdfConverter.java

```
@@ -0,0 +1,122 @@
+/*
+ * Created on 11/15/17.
+ * Written by Islam Salah with assistance from members of Blink22.com
+ */
+
+package android.print;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.Handler;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import java.io.File;
+
+/**
+ * Converts HTML to PDF.
+ * <p>
+ * Can convert only one task at a time, any requests to do more conversions before
+ * ending the current task are ignored.
+ */
+public class PdfConverter implements Runnable {
+
+    private static final String TAG = "PdfConverter";
+    private static PdfConverter sInstance;
+
+    private Context mContext;
+    private String mHtmlString;
+    private File mPdfFile;
+    private PrintAttributes mPdfPrintAttrs;
+    private boolean mIsCurrentlyConverting;
+    private WebView mWebView;
+
+    private PdfConverter() {
+    }
+
+    public static synchronized PdfConverter getInstance() {
+        if (sInstance == null)
+            sInstance = new PdfConverter();
+
+        return sInstance;
+    }
+
+    @Override
+    public void run() {
+        if (mContext == null || mHtmlString == null || mPdfFile == null) return;
+
+        mWebView = new WebView(mContext);
+        mWebView.setWebViewClient(new WebViewClient() {
+            @Override
+            public void onPageFinished(WebView view, String url) {
+                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) return;
```

Might be more appropriate to throw an exception on initialization if the Android version does not support it. This would make it much more obvious to users of this class that PDF conversion is not supported on older versions, rather than failing silently in that case.

---

hi @khaullen , I agree